app/code/Magento/SalesSequence/Model/Builder.php

```diff
@@ -264,4 +274,36 @@ public function create()
         }
         $this->data = array_flip($this->pattern);
     }
+
+    /**
+     * Deletes all sequence linked entites
+     *
+     * @param $storeId
+     *
+     * @return void
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function deleteByStoreId($storeId)
```

Comment:
Hi @Bartlomiejsz!
Looks like the `Builder` class is not the proper place for this `delete` method.
Could you please move this logic into a new separate class `Magento\SalesSequence\Model\DeleteByStore` that provides only one public method `execute(StoreInterface $store)`.

Please check the `\Magento\CustomerGraphQl\Model\Customer\ChangeSubscriptionStatus` as an example.

Thank you!

Reply:
Hi @dmytro-ch, done :)